packages/waku/src/router/create-pages.ts

```diff
@@ -150,19 +150,20 @@ export type CreateLayout = <Path extends string>(
 
 type Method = 'GET' | 'POST' | 'PUT' | 'DELETE';
 
+type ApiHandler = (req: Request) => Promise<Response>;
+
 export type CreateApi = <Path extends string>(
   params:
     | {
-        path: Path;
         mode: 'static';
+        path: Path;
         method: 'GET';
-        handler: (req: Request) => Promise<Response>;
+        handler: ApiHandler;
       }
     | {
-        path: Path;
         mode: 'dynamic';
-        method: Method;
-        handler: (req: Request) => Promise<Response>;
+        path: Path;
+        handlers: Partial<Record<Method, ApiHandler>>;
```

Comment:
Why do you want `Partial`?

Reply:
I think this example explains it better than I can with words

https://tsplay.dev/wOgPpw

Reply:
Oh, yeah. I miss that...